app/components/primer/alpha/navigation_list.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+module Primer
+  module Alpha
+    # `NavigationList` provides a simple way to render side navigation, i.e. navigation
+    # that appears to the left or right side of some main content.
+    # `NavigationList` contains a list of navigation items (links) and corresponding leading
+    # and trailing visuals such as icons. Items can themselves have sub-items.
+    # Finally, `NavigationList` supports sections, which are logical groups of items with
+    # a section header.
+    class NavigationList < Primer::Component
+      # Top-level nav items shown above all sections.
+      #
+      # @param component_klass [Class] A custom component class to use instead of the default Item class.
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_many :items, lambda { |component_klass: Item, **system_arguments|
```

Comment:
Do we need this `component_klass:` option? I don't see it used anywhere and I think we had debated leaving it out at one point.